native/src/main/java/io/xlibb/pipe/Pipe.java

```diff
@@ -181,43 +193,48 @@ public static Object consumeStream(BObject pipe, BDecimal timeout, BTypedesc typ
     }
 
     public static BError produce(Environment env, BObject pipe, Object event, BDecimal timeout) {
-        long timeoutInMillis;
         try {
-            timeoutInMillis = getTimeoutInMillis(timeout);
+            long timeoutInMillis = getTimeoutInMillis(timeout);
+            BHandle handle = (BHandle) pipe.get(NATIVE_PIPE_OBJECT);
+            Pipe nativePipe = (Pipe) handle.getValue();
+            Future future = env.markAsync();
+            Callback observer = new Callback(future, nativePipe.getConsumer(), nativePipe.getTimeKeeper(),
+                    nativePipe.getProducer());
+            nativePipe.asyncProduce(observer, event, timeoutInMillis);
+            return null;
         } catch (BError bError) {
-            return createError("Invalid produce timeout value provided", bError);
+            return createError(INVALID_TIMEOUT_ERROR, bError);
+        } catch (Exception e) {
+            return createError(e.getMessage());
```

Comment:
Don't we have to complete these errors with ballerina futures?

Reply:
Can I know the reason behind that approach?